### PR TITLE
New EventEnvelope including Offset rather than Long #21722

### PR DESCRIPTION
**Pull request by johanandren**
Refs #21722

### CODE REVIEW

File: akka-persistence-query/src/main/scala/akka/persistence/query/EventEnvelope.scala [hunk 1]
```diff
@@ -12,3 +12,14 @@ final case class EventEnvelope(
   persistenceId: String,
   sequenceNr:    Long,
   event:         Any)
+
+/**
+ * Event wrapper adding meta data for the events in the result stream of
+ * [[akka.persistence.query.scaladsl.EventsByTagQuery2]] query, or similar queries.
+ */
+// TODO: Rename it to EventEnvelope in Akka 2.5
```
Comment (ktoso) [inline akka-persistence-query/src/main/scala/akka/persistence/query/EventEnvelope.scala]: + a ticket for it?

Comment (johanandren) [inline akka-persistence-query/src/main/scala/akka/persistence/query/EventEnvelope.scala]: I think there is one already, as this was copied from the #21615 changes, but I will double check.

Comment (johanandren) [inline akka-persistence-query/src/main/scala/akka/persistence/query/EventEnvelope.scala]: Nope, added #21724

**Comment by ktoso** [comment]
LGTM, the 2 because compatibility right?

**Comment by johanandren** [comment]
Correct

**Comment by drewhk** [comment]
EventEnvelope2? :(

**Comment by johanandren** [comment]
Yeah, we will replace the non-numbered ones in 2.5, see https://github.com/akka/akka/pull/21615#discussion_r81776780 for original discussion

**Review by 2m — APPROVED** [review]
`EventEnvelopeV2` might have looked better, but it is too late because of the other `...2` classes.

Anyway, as 2.5 is not that far off, LGTM.

**Pull request merged**